src/react-vite/05-routing/routing.md

```diff
@@ -12,130 +12,294 @@ user interface in response to URL changes.
+
+When you define nested routes in your route configuration above, you don't
+immediately specify where in the parent component's JSX the child components
+should appear. Instead, you use `Outlet` in the parent component's JSX as a
+placeholder for where the matched child route component should be rendered.
+
+In the example below, the child component (`<AboutPage>` or `<HomePage>` from
+our `createBrowserRouter` example) would be rendered inside the `<main>` element:
+
+```tsx
+const Layout = () => {
+  return (
+    <>
+      <header>My awesome site</header>
+      <main>
+        <Outlet /> {/* Child routes will render here */}
+      </main>
+      <footer>©</footer>
+    </>
+  );
+};
+```
+@highlight 6
 
 ### Setup
 
-TODO
+To get starting with React Router in our application,
+let’s install `react-router-dom` with npm:
+
+✏️ Run:
 
 ```shell
 npm install react-router-dom@6
 ```
 
+✏️ Update **src/App.tsx** to be:
+
 @diff ../../../exercises/react-vite/04-components/02-solution/src/App.tsx ../../../exercises/react-vite/05-routing/01-problem/src/App.tsx only
 
+✏️ Update **src/main.tsx** to be:
+
 @diff ../../../exercises/react-vite/04-components/02-solution/src/main.tsx ../../../exercises/react-vite/05-routing/01-problem/src/main.tsx only
 
 ### Verify
 
+✏️ Update **src/App.test.tsx** to be:
+
 TODO
 
-### Exercise: Add routes and an Outlet
+### Exercise 1
 
-TODO
+Create routes for the `<Home>` component and `<RestaurantList>` component.
+When the route is `""`, the `<Home>` component should display, and when
+the route is `/restaurants` then the `<RestaurantList>` component should
+display. These changes should be made in `src/App.tsx` and `src/main.tsx`.
+
+<strong>Having issues with your local setup?</strong> You can use either [StackBlitz](https://stackblitz.com/fork/github/bitovi/academy/tree/main/exercises/react-vite/05-routing/01-problem?file=App.tsx) or [CodeSandbox](https://codesandbox.io/p/devbox/github/bitovi/academy/tree/main/exercises/react-vite/05-routing/01-problem?file=App.tsx) to do this exercise in an online code editor.
 
 ### Solution
 
 <details>
 
 <summary>Click to see the solution</summary>
 
-TODO
+✏️ Update **src/App.tsx** to be:
 
 @diff ../../../exercises/react-vite/05-routing/01-problem/src/App.tsx ../../../exercises/react-vite/05-routing/01-solution/src/App.tsx only
 
+✏️ Update **src/main.tsx** to be:
+
 @diff ../../../exercises/react-vite/05-routing/01-problem/src/main.tsx ../../../exercises/react-vite/05-routing/01-solution/src/main.tsx only
 
+<strong>Having issues with your local setup?</strong> See the solution in [StackBlitz](https://stackblitz.com/fork/github/bitovi/academy/tree/main/exercises/react-vite/05-routing/01-solution?file=App.tsx) or [CodeSandbox](https://codesandbox.io/p/devbox/github/bitovi/academy/tree/main/exercises/react-vite/05-routing/01-solution?file=App.tsx).
+
 </details>
 
-## Objective 2: Links and useMatch
+## Objective 2: Creating links between pages
```

Comment:
At Medline I encountered developers who didn't really understand the different between `<a>` and `<Link>`; they would use an `<a>` tag to navigate without realizing they were reloading the entire page. It might be worth explicitly pointing out how`<a>` is different.

Reply:
That’s great insight, thanks! I’m gonna merge this in as-is, could you open up a PR to add that explanation please?